bindings/go/runtime/json_schema_generator.go

```diff
@@ -0,0 +1,31 @@
+package runtime
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/invopop/jsonschema"
+)
+
+// GenerateJSONSchemaForType takes a Type and uses reflection to generate a JSON Schema representation for it.
+// It will also use the correct type representation as we don't marshal the type in object format.
+func GenerateJSONSchemaForType(obj Typed) ([]byte, error) {
+	r := &jsonschema.Reflector{
+		Mapper: func(i reflect.Type) *jsonschema.Schema {
+			if i == reflect.TypeOf(Type{}) {
```

Comment:
please turn if statement around. Also is it intended that nil is returned if there is no match?

Reply:
Normally, I would agree. But here it's explicit that we do something different in case of this type instead of returning `nil` in case it's NOT that type. 🤔  I think this rather explicitness, reads better.

That said, if you prefer it the other way around I'm okay with that. I'm not dying on this hill. :D 

Yes, the intended use is to return nil, if you don't care about the type:

```go
		if t := r.Mapper(t); t != nil {
			return t
		}
                ... // normal flow
```